stratum/hal/bin/barefoot/README.md

```diff
@@ -233,6 +233,71 @@ journalctl -u stratum_bf.service
 
 See [gNMI CLI](/stratum/tools/gnmi/README.md)
 
+## Stratum-Bf P4DeviceConfig format and the BfPipelineBuilder
+
+Stratum supports two different device configuration formats for pushing the P4
+pipeline over P4Runtime. The older binary packing used by [PI](https://github.com/p4lang/PI)
+and a newer more flexible protobuf based format ([bf.proto](stratum/hal/lib/barefoot/bf.proto)).
+You can use the device config builder to generate it:
+
+```bash
+bazel run //stratum/hal/bin/barefoot:bf_pipeline_builder -- \
+    -p4c_conf_file=/path/to/bf-p4c/compiler/output.conf \
+    -bf_pipeline_config_binary_file=$PWD/device_config.pb.bin
+```
+
+The tool is also available as a [Docker image](https://hub.docker.com/repository/docker/stratumproject/stratum-bf-pipeline-builder):
+
+```bash
+docker run --rm -v $PWD:$PWD stratumproject/stratum-bf-pipeline-builder:latest \
+    -p4c_conf_file=./output.conf \
+    -bf_pipeline_config_binary_file=./device_config.pb.bin
+```
+
+The output goes into the `p4_device_config` field of the P4Runtime
+`ForwardingPipelineConfig` message as usual.
+
+To inspect a `BfPipelineConfig` message, the tool provides an unpacking mode,
+which recreates the original compiler output files:
+
+```bash
+bazel run //stratum/hal/bin/barefoot:bf_pipeline_builder -- \
+    -bf_pipeline_config_binary_file=$PWD/device_config.pb.bin \
+    -unpack_dir=$PWD
+```
+
+### bf-p4c Archive Format
+
+You can enable a special archive format by passing the
+`-incompatible_enable_p4_device_config_tar` flag when starting Stratum.
+
+*Note: Support for this format may disappear at any time.*
+
+The format accepts a tar archive of the bf-p4c compiler output:
+
+```bash
+mkdir -p /tmp/p4out
+bf-p4c <options...> \
```

Comment:
use archive flag

Reply:
The `--archive` flags does NOT pack everything we need!
In particular, the `tofino.bin` files are missing: https://forum.barefootnetworks.com/hc/en-us/community/posts/360046930333-Missing-tofino-bin-in-archive-produced-by-tofino-p4c

Reply:
got it... looks like you are about 4 months ahead of me on this